app/controllers/comparisons/base_controller.rb

```diff
@@ -156,5 +163,22 @@ def included_schools
       filter = filter.accessible_by(current_ability, :show) unless include_invisible
       filter.pluck(:id)
     end
+
+    def switch_layout
+      params[:group] == 'true' ? 'dashboards' : 'application'
+    end
+
+    def set_school_group_vars
+      @school_group_layout = params[:group] == 'true'
```

Comment:
This looks like a mixed bag of things to do all in one method. I'd be thinking about splitting it out into functional groups